frontend/utils/identity.ts

```diff
@@ -0,0 +1,11 @@
+import { Principal } from "@dfinity/principal";
+
+export function validatePrincipal(principal: string): boolean {
+  try {
+    Principal.fromText(principal);
```

Comment:
i think dfinity already provides a validation for the principal. please verify

Reply:
We need to validate input as it is being typed. This is the best option for this case so far.